web/src/lib/components/shared-components/profile-image-cropper.svelte

```diff
@@ -0,0 +1,55 @@
+<script lang="ts">
+  import { AssetResponseDto, api } from '@api';
+  import { createEventDispatcher } from 'svelte';
+  import PhotoViewer from '../asset-viewer/photo-viewer.svelte';
+  import BaseModal from './base-modal.svelte';
+  import Button from '../elements/buttons/button.svelte';
+  const dispatch = createEventDispatcher();
+  export let asset: AssetResponseDto;
+  export let publicSharedKey = '';
+  import domtoimage from 'dom-to-image';
+  import { notificationController, NotificationType } from './notification/notification';
+  import { handleError } from '$lib/utils/handle-error';
+
+  let profilePicture: HTMLDivElement;
+
+  const handleSetProfilePicture = async () => {
+    const div = profilePicture.childNodes[0];
+    const blob = await domtoimage.toBlob(div);
+    const file = new File([blob], 'profile-picture.png', { type: 'image/png' });
+    try {
+      await api.userApi.createProfileImage({ file });
+      dispatch('close');
```

Comment:
Should this also be dispatched when an error is thrown?

Reply:
I think when `await api.userApi.createProfileImage({file})` throws an error, the execution will directly go into the catch block and not dispatch the close event. Is that not the case?

Reply:
I believe so, I was asking if it should be dispatched even in the face of an error. I suppose not?